wit/couchbase/types.wit

```diff
@@ -0,0 +1,210 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// General types that can be used throughout the couchbase interface
+interface types {
+  /// ID of a Document
+  ///
+  /// Note that Document IDs are expected to be:
+  ///   - Assigned at the application level
+  ///   - Be valid UTF-8 at the application level
+  ///   - Be no longer than 250 bytes (not characters)
+  type document-id = string;
+
+  /// Path to a subdocument inside an existing document
+  type subdocument-path = string;
+
+  /// The span of a request, normally used when performing/enabling tracing
+  type request-span = string;
+
+  /// A string that is properly formatted JSON
+  type json-string = string;
+
+  /// Document field name (ex. to include in a search)
+  type document-field-name = string;
+
+  /// Collection name to include in a search
+  type collection-name = string;
+
+  /// Name of a search index
+  type search-index-name = string;
+
+  /// Whether to enable replica reads for the request
+  enum replica-read-level {
+    off,
+    on,
+  }
+
+  /// Direction in which to perform sorting
+  enum sort-direction {
+    asc,
+    desc,
+  }
+
+  /// Port of https://pkg.go.dev/time#Time
+  record time {
+    /// Offsets are assumed to be against the western hemisphere (GMT)
+    /// i.e. -14 -> 0 -> +14 UTC
+    offset: s8,
+    year: s32,
+    month: u8,
+    day: u8,
+    // 24hr time (i.e. 6PM is represented as `18`)
+    hour: u8,
+    minute: u8,
+    second: u8,
+    milliseconds: u32,
+    nanoseconds: u32,
+  }
+
+  /// A keyspace-path that identifies a collection
+  record collection {
+    /// Bucket the collection belongs to
+    bucket: option<string>,
+    /// Scope of the collection (ex. "_default")
+    scope: option<string>,
+    /// Name of the collection (which may be a simple name, if only this value is specified)
```

Comment:
I don't understand how it will behave if I only pass the name of the collection ?

Reply:
Ah so here I was thinking that the "simple" name would work -- `_default` is the scope if not specified right? though the default bucket name is not clear to me. Maybe I'm misunderstanding the Couchbase API here

Reply:
Ah I see now, the default bucket is really not meant to be used anywhere in production -- I've required a specific bucket name, but the scope can still be optional (and I've clarified that `_default` is used if not specified!)